zanata-war/src/main/java/org/zanata/dao/ProjectDAO.java

```diff
@@ -311,6 +286,34 @@ private FullTextQuery getTextQuery(@Nonnull String searchQuery,
         return entityManager.createFullTextQuery(booleanQuery, HProject.class);
     }
 
+    /**
+     * Build BooleanQuery on single lucene field by splitting searchQuery with
+     * white space.
+     *
+     * @param searchQuery
+     *            - query string, will replace hypen with space and escape
+     *            special char
+     * @param field
+     *            - lucene field
+     */
+    private BooleanQuery buildSearchFieldQuery(@Nonnull String searchQuery,
+        @Nonnull String field) throws ParseException {
+        BooleanQuery query = new BooleanQuery();
+
+        //replace hyphen with space for lucene search
+        searchQuery = searchQuery.replace("-", " ");
```

Comment:
we allow ".", "-" and "_" in project slug. Not sure whether you need to replace the other two as well here.